src/ShallowWrapper.js

```diff
@@ -350,8 +350,11 @@ export default class ShallowWrapper {
     const handler = this.prop(propFromEvent(event));
     if (handler) {
       withSetStateAllowed(() => {
-        // TODO(lmr): create/use synthetic events
         // TODO(lmr): emulate React's event propagation
+        if (!args.length && window) {
+          args.push(new window.Event(event));
```

Comment:
Manually create an event object. The behavior of `shallow` should not differ based on the presence or absence of browser globals.